Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -441,10 +441,13 @@ fileprivate extension Array<AnnotatedTestItem> {
         // as the root item.
         if rootItem.isExtension && !item.isExtension {
           var newItem = item
-          newItem.testItem.children += rootItem.testItem.children
+          newItem.testItem.children = (newItem.testItem.children + rootItem.testItem.children).deduplicateIds()
           rootItem = newItem
+        } else if rootItem.testItem.children.isEmpty && item.testItem.children.isEmpty {
+          itemDict[item.testItem.ambiguousTestDifferentiator] = item
+          continue
```

Comment:
I got a little confused by this branch being between the two other if-branches that are concerned with merging two test items. I think it would read easier if this is the first branch in the `if` statement.

Reply:
The order matters here. If the first condition is true we don't want to evaluate this one.

Reply:
Can we have a test item that’s an extension without any members?

Reply:
I need a bit of clarification, do you mean a test item that has no children? 

If so, it’s possible to have an empty extension which would be represented as a separate test item initially, but the rest of this function does the work of merging the extension and the original declaration.